File: Code/Chapter 13/VB.NET/HandWrittenUDF/UserDefinedFunctions.vb
Imports System
Imports System.Data.SqlTypes
Imports Microsoft.SqlServer.Server

Public Class UserDefinedFunctions
   <Microsoft.SqlServer.Server.SqlFunction()> _
   Public Shared Function GetRandomNumber() As SqlInt32
      Dim rnd As Random = New Random
      Return rnd.Next()
   End Function
End Class

File: Code/Chapter 13/C#/HandWrittenUDF/UserDefinedFunctions.cs
using System;
using System.Data.SqlTypes;
using Microsoft.SqlServer.Server;

namespace HandWrittenUDF
{
   public class UserDefinedFunctions
   {
      [Microsoft.SqlServer.Server.SqlFunction]
      public static SqlInt32 GetRandomNumber()
      {
         Random rnd = new Random();
         return rnd.Next();
      }
   }
}

// SQL SCRIPT
//Create Assembly HandWrittenUDF
//from
//'C:\PlayArea\SqlServerProject\HandWrittenUDF\bin\Debug\HandWrittenUDF.dll'
//GO

//Create Function GetRandomNumber ()
//Returns Int
//As
//External Name
//HandWrittenUDF.[HandWrittenUDF.UserDefinedFunctions].GetRandomNumber
//Go

//Select dbo.GetRandomNumber()
//Go